src/System.Net.Http/src/System/Net/Http/Headers/HeaderUtilities.cs

```diff
@@ -63,6 +63,48 @@ internal static void SetQuality(ObjectCollection<NameValueHeaderValue> parameter
             }
         }
 
+        // Encode a string using RFC 5987 encoding.
+        // encoding'lang'PercentEncodedSpecials
+        internal static string Encode5987(string input)
+        {
+            string output;
+            Encode5987(input, out output);
+
+            return output;
+        }
+
+        internal static bool Encode5987(string input, out string output)
```

Comment:
I recommend you rename this method.  It isn't clear what the 'bool' return value is.  It like like 'TryEncode5987' where it succeeds or fails?